src/internal/kopia/base_finder.go

```diff
@@ -304,30 +307,108 @@ func (b *baseFinder) findBasesInSet(
 		// If we've made it to this point then we're considering the backup
```

Comment:
I can't comment on them directly but lines 283/286-302/305 should be changed to just `continue`. We can't add anything to the assist snapshot set anymore unless it fits the (new) definition of an assist base

The key to remember with this code is that it's selecting both the backups that will be used to source details during details merging and the snapshots that will be used either for hierarchy merging (merge bases) or passed to kopia to check for cached files (assist base)